internal/fetchdatasource/fetchdatasource.go

```
@@ -393,3 +394,7 @@ func (ds *FetchDataSource) Search(ctx context.Context, q string, opts internal.S
 
 	return results, nil
 }
+
+func (ds *FetchDataSource) GetLLPkgConfig(ctx context.Context, modulePath, version string) (*llpkgcfg.LLPkgConfig, error) {
+	return nil, nil
+}
```

This implementation returns `nil, nil`, which may mislead into thinking it indicates "success with no data" rather than "feature not supported."

---

add the module path verify at [f5235f8](https://github.com/goplus/pkgsite/pull/21/commits/f5235f85d2e4c49806e4bef67a0dd0c66be3cf35)

---

```go
// GetLLPkgConfig gets the LLPkgConfig file through module path match LLPkg rules.
func (ds *FetchDataSource) GetLLPkgConfig(ctx context.Context, modulePath, version string) (*llpkgcfg.LLPkgConfig, error) {
	if llpkg.IsLLPkgModule(modulePath) {
		return nil, nil
	}
	return nil, fmt.Errorf("%s@%s is not a LLPkg module", modulePath, version)
}
```
This update might still have issues. If FetchDataSource acts as the data source, even when it is a LLPkg, it cannot fetch the `llpkgcfg.LLPkgConfig` and returns `nil, nil`, which implies there are no errors. However, the current `(ds *FetchDataSource) GetLLPkgConfig` actually does not implement the functionality to read the config from the module path. In this case, it should return an error (if this logic is not to be implemented at present) to indicate that the `GetLLPkgConfig` is not supported in `FetchDataSource`, otherwise it might mislead some callers who use `FetchDataSource` to fetch the configuration file.

---

done at [3a26209](https://github.com/goplus/pkgsite/pull/21/commits/3a262098961edd9237b7799f5f8441b07d1a3513)
Now `GetLLPkgConfig` will always return err in `FetchDataSource`
`GetLLPkgConfig` in `postgres` will also check if the moudle is a LLPkg

---

```go
func (ds *FetchDataSource) GetLLPkgConfig(ctx context.Context, modulePath, version string) (*llpkgcfg.LLPkgConfig, error) {
	if llpkg.IsLLPkgModule(modulePath) {
		return nil, fmt.Errorf("%s@%s is not a LLPkg module", modulePath, version)
	}
	return nil, fmt.Errorf("GetLLPkgConfig is not implemented in FetchDataSource")
}
```
`todo:GetLLPkgConfig current is not implemented in FetchDataSource`,is better。
 both of `IsLLPkg` & `NotLLPkg` 's situation need  same todo error,current `fmt.Errorf("%s@%s is not a LLPkg module", modulePath, version)` in `IsLLPkg` also lead misunderstanding

---

done at [75d13a5](https://github.com/goplus/pkgsite/pull/21/commits/75d13a5768d89030c908c1b1e8a8343f2c09ad2b)
Now these behavior will be the consistent both at error return & code logic

---

LGTM